applications/product/groovyScripts/product/cost/CostServices.groovy

```diff
@@ -495,15 +497,14 @@ def getProductAverageCost() {
 
 /**
  * Formula that creates a cost component equal to a percentage of total product cost
- * @return
  */
-def productCostPercentageFormula() {
+Map productCostPercentageFormula() {
     Map result = success()
     GenericValue productCostComponentCalc = parameters.productCostComponentCalc
     GenericValue costComponentCalc = parameters.costComponentCalc
-    Map inputMap = [productId: productCostComponentCalc.productId, currencyUomId: parameters.currencyUomId, costComponentTypePrefix: parameters.costComponentTypePrefix]
-    Map serviceResult = run service: "getProductCost", with: inputMap
-    BigDecimal productCost = serviceResult.productCost
+    Map inputMap = [productId: productCostComponentCalc.productId, currencyUomId: parameters.currencyUomId,
+                    costComponentTypePrefix: parameters.costComponentTypePrefix]
+    run service: 'getProductCost', with: inputMap
     // set field="productCostAdjustment" value="${parameters.baseCost * costComponentCalc.fixedCost}" type="BigDecimal"/
```

Comment:
Should we keep this comment and if so why?

Reply:
Like Dan said in another comment, i had to refrain to change a lot of things during rule fixes, let's try to avoid mixing objectives in this pull-request :)

Reply:
OK